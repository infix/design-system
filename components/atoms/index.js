import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
