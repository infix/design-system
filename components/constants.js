/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
