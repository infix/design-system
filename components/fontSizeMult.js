/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
