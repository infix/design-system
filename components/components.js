import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
