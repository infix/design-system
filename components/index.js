import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
