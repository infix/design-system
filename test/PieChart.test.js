/* global describe, test, expect */
import React from 'react'
import { renderShallowJSON } from './index'
import { PieChart } from '../components'

describe('PieChart', () => {
  test('renders', () => {
    const json = renderShallowJSON(
      <PieChart />
    )
    expect(json).toMatchSnapshot()
  })
})
