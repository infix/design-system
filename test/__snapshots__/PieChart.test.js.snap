// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart renders 1`] = `
<svg
  height={400}
  width={400}
>
  <ChartTitle />
  <ChartTitle
    subtitle={true}
  />
  <g
    transform="translate(50, 0)"
  >
    <VictoryPie
      colorScale={
        Array [
          "#0588cb",
          "#e64980",
          "#ffa94d",
          "#c0eb75",
          "#6741d9",
        ]
      }
      containerComponent={
        <VictoryContainer
          portalComponent={
            <Portal
              groupComponent={<g />}
            />
          }
          responsive={true}
        />
      }
      data={
        Array [
          Object {
            "x": "A",
            "y": 1,
          },
          Object {
            "x": "B",
            "y": 2,
          },
          Object {
            "x": "C",
            "y": 3,
          },
          Object {
            "x": "D",
            "y": 1,
          },
          Object {
            "x": "E",
            "y": 2,
          },
        ]
      }
      dataComponent={<Slice />}
      groupComponent={<g />}
      height={400}
      innerRadius={50}
      labelComponent={<PieLabelComponent />}
      labelRadius={80}
      padding={
        Object {
          "bottom": 80,
          "left": 80,
          "right": 80,
          "top": 80,
        }
      }
      standalone={true}
      theme={
        Object {
          "area": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "#252525",
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
                "textAnchor": "middle",
              },
            },
            "width": 450,
          },
          "axis": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "axis": Object {
                "fill": "transparent",
                "stroke": "#252525",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": 1,
              },
              "axisLabel": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 25,
                "stroke": "transparent",
                "textAnchor": "middle",
              },
              "grid": Object {
                "fill": "transparent",
                "pointerEvents": "none",
                "stroke": "transparent",
              },
              "tickLabels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
              },
              "ticks": Object {
                "fill": "transparent",
                "size": 1,
                "stroke": "transparent",
              },
            },
            "width": 450,
          },
          "bar": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "#252525",
                "padding": 8,
                "strokeWidth": 0,
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
              },
            },
            "width": 450,
          },
          "candlestick": Object {
            "candleColors": Object {
              "negative": "#252525",
              "positive": "#ffffff",
            },
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "data": Object {
                "stroke": "#252525",
                "strokeWidth": 1,
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
                "textAnchor": "middle",
              },
            },
            "width": 450,
          },
          "chart": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "width": 450,
          },
          "errorbar": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "transparent",
                "stroke": "#252525",
                "strokeWidth": 2,
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
                "textAnchor": "middle",
              },
            },
            "width": 450,
          },
          "group": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "width": 450,
          },
          "legend": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "gutter": 10,
            "orientation": "vertical",
            "style": Object {
              "data": Object {
                "type": "circle",
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
              },
              "title": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 5,
                "stroke": "transparent",
              },
            },
            "titleOrientation": "top",
          },
          "line": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "transparent",
                "stroke": "#252525",
                "strokeWidth": 2,
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
                "textAnchor": "middle",
              },
            },
            "width": 450,
          },
          "pie": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 400,
            "padding": 50,
            "style": Object {
              "data": Object {
                "padding": 10,
                "stroke": "transparent",
                "strokeWidth": 1,
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 20,
                "stroke": "transparent",
              },
            },
            "width": 400,
          },
          "scatter": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "#252525",
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 10,
                "stroke": "transparent",
                "textAnchor": "middle",
              },
            },
            "width": 450,
          },
          "stack": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "width": 450,
          },
          "tooltip": Object {
            "cornerRadius": 5,
            "flyoutStyle": Object {
              "fill": "#f0f0f0",
              "pointerEvents": "none",
              "stroke": "#252525",
              "strokeWidth": 1,
            },
            "pointerLength": 10,
            "style": Object {
              "fill": "#252525",
              "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
              "fontSize": 14,
              "letterSpacing": "normal",
              "padding": 5,
              "pointerEvents": "none",
              "stroke": "transparent",
              "textAnchor": "middle",
            },
          },
          "voronoi": Object {
            "colorScale": Array [
              "#252525",
              "#525252",
              "#737373",
              "#969696",
              "#bdbdbd",
              "#d9d9d9",
              "#f0f0f0",
            ],
            "height": 300,
            "padding": 50,
            "style": Object {
              "data": Object {
                "fill": "transparent",
                "stroke": "transparent",
                "strokeWidth": 0,
              },
              "flyout": Object {
                "fill": "#f0f0f0",
                "pointerEvents": "none",
                "stroke": "#252525",
                "strokeWidth": 1,
              },
              "labels": Object {
                "fill": "#252525",
                "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Ser­avek', 'Trebuchet MS', sans-serif",
                "fontSize": 14,
                "letterSpacing": "normal",
                "padding": 5,
                "pointerEvents": "none",
                "stroke": "transparent",
                "textAnchor": "middle",
              },
            },
            "width": 450,
          },
        }
      }
      width={400}
    />
  </g>
  <g
    transform="translate(25, 250)"
  >
    <OONILogoWordmarkHoriz
      width={100}
    />
  </g>
</svg>
`;
